From c5cb2c37b21de487b2224d7999749bdc1f4442c2 Mon Sep 17 00:00:00 2001
From: cyrus <linchaolong.dev@gmail.com>
Date: Wed, 8 Apr 2026 14:56:07 +0800
Subject: [PATCH 05/11] thread gmain

---
 src/obfuscate_agent_symbols.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/obfuscate_agent_symbols.py b/src/obfuscate_agent_symbols.py
index 76cca7f0..a3e48c0f 100644
--- a/src/obfuscate_agent_symbols.py
+++ b/src/obfuscate_agent_symbols.py
@@ -42,9 +42,14 @@ if __name__ == "__main__":
                 log_color(f"[*] Patching section name={section.name} offset={hex(section.file_offset + addr)} orig:{patch_str} new:{''.join(list(patch_str)[::-1])}")
                 binary.patch_address(section.file_offset + addr, patch)
 
+    binary.write(input_file)
+
     # thread gum_js_loop
     random_name = "".join(random.sample(random_charset, 11))
     log_color(f"[*] Patch `gum-js-loop` to `{random_name}`")
     os.system(f"sed -b -i s/gum-js-loop/{random_name}/g {input_file}")  # 把 gum-js-loop 替换成随机字符串
 
-    binary.write(input_file)
+    # thread gmain
+    random_name = "".join(random.sample(random_charset, 5))
+    log_color(f"[*] Patch `gmain` to `{random_name}`")
+    os.system(f"sed -b -i s/gmain/{random_name}/g {input_file}")
-- 
2.34.1

